Application/EdFi.Ods.SwaggerUI/Startup.cs

```diff
@@ -67,11 +77,56 @@ void AppSettingsDelegate(IApplicationBuilder app)
                 app.UseExceptionHandler("/Error");
             }
 
-            app.Map("/appSettings.json", AppSettingsDelegate);
+            app.Map(
+                "/appSettings.json", builder =>
+                {
+                    builder.Run(
+                        async context =>
+                        {
+                            context.Response.ContentType = "application/json";
 
-            app.UseEdFiSwaggerUI();
+                            await context.Response.WriteAsync(
+                                JsonSerializer.Serialize(
+                                    new
+                                    {
+                                        WebApiVersionUrl = webApiUrl,
+                                        RoutePrefix = _routePrefix ?? "swagger"
+                                    }));
+                        });
+                });
+
+            logger.LogInformation($"RoutePrefix = '{_routePrefix}'");
+            logger.LogInformation($"WebApiUrl = '{webApiUrl}'");
+            logger.LogInformation($"UseReverseProxyHeaders = '{_useReverseProxyHeaders}'");
+            logger.LogInformation($"PathBase = '{_pathBase}'");
+            logger.LogInformation($"SwaggerStyleSheetPath = '{swaggerStyleSheetPath}'");
+
+            app.UseSwaggerUI(
+                options =>
+                {
+                    Configuration.Bind("SwaggerUIOptions", options);
+
+                    options.DocumentTitle = "Ed-Fi ODS API Documentation";
+
+                    options.OAuthScopeSeparator(" ");
+                    options.OAuthAppName(Assembly.GetExecutingAssembly().GetName().Name);
+                    options.DocExpansion(DocExpansion.None);
+                    options.DisplayRequestDuration();
+                    options.ShowExtensions();
+                    options.EnableValidator();
+                    options.InjectStylesheet(swaggerStyleSheetPath);
+
+                    options.IndexStream = ()
+                        => GetType().Assembly.GetManifestResourceStream("EdFi.Ods.SwaggerUI.Resources.Swashbuckle_index.html");
+
+                    options.ConfigObject.AdditionalItems["WebApiVersionUrl"] = webApiUrl;
+
+                    // default behavior requires "swagger"
+                    options.RoutePrefix = _routePrefix ?? "swagger";
```

Comment:
I think we actually should have used `SwaggerUIOptions:RoutePrefix` originally as it is a part of the `SwaggerUIOptions` class. If we set it this way can we can support both options? RoutePrefix already defaults to "swagger".
```suggestion
                    options.RoutePrefix = _routePrefix ?? options.RoutePrefix;
```

Reply:
I tried this originally, and this does not work as the bind statement changes the value to null, (unless we define the value). The original code also always set the value to correct this problem.

Reply:
I agree that using swaggeroptions:routeprefix is the correct solution over the original implementation of routeprefix. The question becomes, should we make this change, and should we adjust octopus accordingly?